### PR TITLE
Update the Number Lookup preview to include Number Format features

### PR DESCRIPTION
**Pull request by ericasp16**
# Description
Update the Number Lookup public preview to include the Number Format features we've been preparing for GA.

https://github.com/Azure/azure-rest-api-specs/pull/27799

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java [hunk 1]
```diff
@@ -695,4 +699,40 @@ public PagedIterable<PhoneNumberOffering> listAvailableOfferings(String countryC
...
+     *
+     * @param phoneNumbers The phone number(s) whose operator information should be searched.
+     * @param includeAdditionalOperatorDetails Modifies the search to include additional fields in the response.
+     *                  Please note: use of this option will affect the cost of the search.
+     * @param context A {@link Context} representing the request context.
+     *
+     * @return A {@link OperatorInformationResult} which contains the results of the search.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<OperatorInformationResult> searchOperatorInformationWithResponse(List<String> phoneNumbers,
```
[the remark this answers is not in the document]
Comment (ericasp16) [reply sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java:729]: it has to be 'With Response' to be consistent with API standards

[the remark this answers is not in the document]
Comment (guoqing2023) [reply sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java:729]: I see. So this is the method include context as an input parameter. Does the previous method needs to have includeAdditionalOperatorDetails as input parameter?

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java [hunk 2]
```diff
@@ -695,4 +699,40 @@ public PagedIterable<PhoneNumberOffering> listAvailableOfferings(String countryC
         context = context == null ? Context.NONE : context;
         return client.listOfferings(countryCode, null, null, phoneNumberType, assignmentType, acceptLanguage, context);
     }
+
+    /**
+     * Searches for operator information for a given list of phone numbers.
```
[the remark this answers is not in the document]
Comment (ericasp16) [reply sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java:704]: I'm going to leave it as-is for two reasons:
1. It's consistent with other overloaded methods in this file
2. The function is basically the same it's just an overload

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericasp16) [reply sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java:729]: According to the feedback I got in the stewardship review these two overrides are fine, they usually just add 'withResponse' to whichever overload takes the most parameters

**Review by guoqing2023 — APPROVED** [review]

File: sdk/communication/azure-communication-phonenumbers/CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
-## 1.2.0-beta.1 (Unreleased)
+## 1.2.0-beta.2 (2024-03-01)
 
 ### Features Added
+- Add support for number lookup
+  - Format only can be returned for no cost
+  - Additional number details can be returned for a cost
 
-### Breaking Changes
+## 1.2.0-beta.1 (2023-09-07)
```
Comment (srnagar) [inline sdk/communication/azure-communication-phonenumbers/CHANGELOG.md:10]: This looks out of order. We have releases in December 2023 below this. These should be sorted in reverse chronological order.

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersServiceVersion.java [hunk 4]
```diff
@@ -12,7 +12,12 @@ public enum PhoneNumbersServiceVersion implements ServiceVersion {
     /**
      * Service version {@code 2021-03-07}.
      */
-    V2021_03_07("2021-03-07");
+    V2021_03_07("2021-03-07"),
+
+    /**
+     * Number Lookup update to public preview {@code 2024-03-01-preview}
+     */
+    V2024_03_01_Preview("2024-03-01-preview");
```
Comment (srnagar) [inline sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersServiceVersion.java:20]:
```suggestion
    V2024_03_01_PREVIEW("2024-03-01-preview");
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**